packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -159,6 +178,140 @@ async function createOutgoingPayment(
   }
 }
 
+function validateAccessLimits(
+  payment: OutgoingPayment,
+  limits: AccessLimits
+): PaymentLimits | undefined {
+  if (
+    (!limits.receiver || payment.receiver === limits?.receiver) &&
+    validateAmountAssets(payment, limits)
+  ) {
+    let paymentInterval: Interval | undefined
+    if (limits.interval)
+      paymentInterval = getInterval(limits.interval, payment.createdAt)
+    if (!limits.interval || !!paymentInterval) {
+      return { ...limits, paymentInterval }
+    }
+  }
+}
+
+function validatePaymentInterval({
+  limits,
+  payment
+}: {
+  limits: PaymentLimits
+  payment: OutgoingPayment
+}): boolean {
+  if (
+    !limits.paymentInterval ||
+    (limits.paymentInterval.start.toMillis() <= payment.createdAt.getTime() &&
+      payment.createdAt.getTime() < limits.paymentInterval.end.toMillis())
+  ) {
+    return true
+  }
+  return false
+}
+
+function validateAmountAssets(
+  payment: OutgoingPayment,
+  limits: AccessLimits
+): boolean {
+  if (
+    limits.sendAmount &&
+    (limits.sendAmount.assetCode !== payment.asset.code ||
+      limits.sendAmount.assetScale !== payment.asset.scale)
+  ) {
+    return false
+  }
+  return (
+    !limits.receiveAmount ||
+    (limits.receiveAmount.assetCode === payment.receiveAmount.assetCode &&
+      limits.receiveAmount.assetScale === payment.receiveAmount.assetScale)
+  )
+}
+
+interface PaymentLimits extends AccessLimits {
+  paymentInterval?: Interval
+}
+
+// "payment" is locked by the "deps.knex" transaction.
+async function validateGrant(
+  deps: ServiceDependencies,
+  payment: OutgoingPayment,
+  grant: Grant
+): Promise<boolean> {
+  const grantAccess = grant.access[0]
+  if (!grantAccess.limits?.sendAmount && !grantAccess.limits?.receiveAmount) {
+    return true
```

Comment:
This returns `true` before `receiver` is checked.
Maybe it should be `if (!grantAccess.limits)` here and move this below `validateAccessLimits`?